moshi/src/main/java/com/squareup/moshi/Util.java

```diff
@@ -18,27 +18,41 @@
 import java.lang.annotation.Annotation;
 import java.lang.reflect.AnnotatedElement;
 import java.lang.reflect.Type;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Set;
 
 final class Util {
-  public static final Annotation[] EMPTY_ANNOTATIONS_ARRAY = new Annotation[0];
-
-  public static final AnnotatedElement NO_ANNOTATIONS = new AnnotatedElement() {
-    @Override public boolean isAnnotationPresent(Class<? extends Annotation> aClass) {
-      return false;
-    }
-    @Override public <T extends Annotation> T getAnnotation(Class<T> tClass) {
-      return null;
-    }
-    @Override public Annotation[] getAnnotations() {
-      return EMPTY_ANNOTATIONS_ARRAY;
-    }
-    @Override public Annotation[] getDeclaredAnnotations() {
-      return EMPTY_ANNOTATIONS_ARRAY;
-    }
-  };
+  public static final Set<Annotation> NO_ANNOTATIONS = Collections.emptySet();
```

Comment:
I did the opposite, changing callers to `Util.NO_ANNOTATIONS`. It's got ~12 uses. I'm open to flip flop again on this.